source/Halibut/Transport/ConnectionManager.cs

```diff
@@ -47,6 +45,30 @@ public async Task<IConnection> AcquireConnectionAsync(ExchangeProtocolBuilder ex
                 return openableConnection;
             }
         }
+        
+        async Task<IConnection> TryGetExistingConnectionAsync(ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken)
+        {
+            await Task.CompletedTask;
+            lock (activeConnections)
```

Comment:
Does this even work? a lock in an async!!

Reply:
hmm I had understood it would use a sync lock, which I am less concerned about since the lock shouldn't be held while doing IO.

We could change it but it does mean the sync code must also change.

Reply:
No ignore, you can't call async inside a lock. My misunderstanding here 👍